pkg/aspect/accessLogsManager.go

```diff
@@ -132,52 +127,33 @@ func (e *accessLogsWrapper) Close() error {
 }
 
 func (e *accessLogsWrapper) Execute(attrs attribute.Bag, mapper expr.Evaluator, ma APIMethodArgs) (*Output, error) {
-	// TODO: would be nice if we could use a mutable.Bag here and could pass it around
-	// labels holds the generated attributes from mapper
-	labels := make(map[string]interface{})
-	for attr, e := range e.inputs {
-		if val, err := mapper.Eval(e, attrs); err == nil {
-			labels[attr] = val
-		}
-		// TODO: throw error on failed mapping?
+	labels, err := evalAll(e.labels, attrs, mapper)
+	if err != nil {
+		return &Output{Code: rpc.INTERNAL}, fmt.Errorf("failed to eval labels for log %s with err: %s", e.name, err)
 	}
 
-	// TODO: better way to ensure timestamp is available if not supplied
-	// in Report() requests.
-	if _, found := labels["timestamp"]; !found {
-		labels["timestamp"] = time.Now()
+	templateVals, err := evalAll(e.templateExprs, attrs, mapper)
+	if err != nil {
+		return &Output{Code: rpc.INTERNAL}, fmt.Errorf("failed to eval template expressions for log %s with err: %s", e.name, err)
 	}
 
-	entry := adapter.LogEntry{
-		LogName: e.logName,
-		Labels:  make(map[string]interface{}),
+	buf := bufferPool.Get().(*bytes.Buffer)
+	if err := e.template.Execute(buf, templateVals); err != nil {
+		buf.Reset()
+		bufferPool.Put(buf)
+		return &Output{Code: rpc.INTERNAL}, err
 	}
+	payload := buf.String()
+	buf.Reset()
+	bufferPool.Put(buf)
 
-	for _, a := range e.attrNames {
-		if val, ok := labels[a]; ok {
-			entry.Labels[a] = val
-			continue
-		}
-		if val, found := attribute.Value(attrs, a); found {
-			entry.Labels[a] = val
-		}
-
-		// TODO: do we want to error for attributes that cannot
-		// be found?
-	}
-
-	if len(entry.Labels) == 0 {
-		// don't write empty access logs
-		return &Output{Code: rpc.OK}, nil
-	}
-
-	buf := new(bytes.Buffer)
-	if err := e.template.Execute(buf, entry.Labels); err != nil {
-		return nil, err
+	entry := adapter.LogEntry{
```

Comment:
We need a way to have access logs reported the constructed string as well as the bits that made up the template (for structured logging).

It sounds like we are saying now that you will have duplicated template_expressions and labels clauses in the case where you want both?

what if we just drop template_expressions all together and require that the expression be written in terms of the labels (if you want an expression)?